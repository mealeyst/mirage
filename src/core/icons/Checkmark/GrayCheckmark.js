import PropTypes from 'prop-types'
import styled from 'styled-components'

import Checkmark from './Checkmark.base'


const GrayCheckmark = styled(Checkmark)`
  stroke: ${props => props.theme.colors.loading}
`

GrayCheckmark.propTypes = {
  theme: PropTypes.shape({
    colors: PropTypes.shape({
      loading: PropTypes.string
    })
  })
}

/** @component */
export default GrayCheckmark
