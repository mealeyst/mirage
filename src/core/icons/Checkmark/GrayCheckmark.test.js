import React from 'react'
import 'jest-styled-components'

import GrayCheckmark from './GrayCheckmark'

const { mountWithTheme } = global

describe('(Styled Component) GrayCheckmark', () => {
  const createGrayCheckmark = (props) => {
    return mountWithTheme(<GrayCheckmark {...props} />)
  }

  test('matching the snapshot', () => {
    expect(createGrayCheckmark())
    .toMatchSnapshot()
  })
})
