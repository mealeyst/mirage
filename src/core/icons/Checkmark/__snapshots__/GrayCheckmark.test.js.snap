// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Styled Component) GrayCheckmark matching the snapshot 1`] = `
.c1 {
  width: 2.6rem;
  fill: none;
  stroke-width: 2;
}

.c1 > polyline {
  stroke-linecap: round;
  stroke-dasharray: 16;
  stroke-dashoffset: 0;
  -webkit-animation-name: gtzAxz;
  animation-name: gtzAxz;
  -webkit-animation-duration: 0.5s;
  animation-duration: 0.5s;
}

.c0 {
  stroke: #BFBFBF;
}

<Styled(Styled(UnstyledCheckmark))>
  <Styled(UnstyledCheckmark)
    className="c0"
    width="2.6rem"
  >
    <UnstyledCheckmark
      className="c0 c1"
      width="2.6rem"
    >
      <svg
        className="c0 c1"
        viewBox="0 0 26.5 26.5"
        width="2.6rem"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polyline
          points="7,14 11,17.7 18.6,9.5"
        />
      </svg>
    </UnstyledCheckmark>
  </Styled(UnstyledCheckmark)>
</Styled(Styled(UnstyledCheckmark))>
`;
